### PR TITLE
0.2.0: optional release-mode asserts, remove {ui}size restrictions, replace bad "reverse" checks

### LINKED ISSUE #5: Conversion from `u32` to `f32` via `cast()` can be inexact
My understanding from the docs is that conversions via `cast()` should always be exact, but this is not currently the case.

```rust
let x: f32 = (!0u32).cast();
println!("{}", !0u32); // prints 4294967295
println!("{}", x); // prints 4294967300
```

### PR DESCRIPTION
**Pull request by dhardy**
Closes #5 and #3.

@BartMassey can you review please?

### CODE REVIEW

**Comment by dhardy** [comment]
Also closes #1. Pending review, I'll release v0.2 tomorrow.

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Changelog
+=========
+
+## unreleased
+
+-   Remove restrictions on isize/usize (#6)
+-   Replace bad revert check (#6)
```
Comment (BartMassey) [inline CHANGELOG.md]: Should be #5 here, I think?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,8 @@
 Changelog
 =========
 
-## unreleased
+## [0.2.0] — 2021-03-20
 
+-   Add feature flags controlling assert behaviour
 -   Remove restrictions on isize/usize (#6)
--   Replace bad revert check (#6)
+-   Fix bad revert checks (#6)
```
Comment (BartMassey) [inline CHANGELOG.md:8]: Should be #5 here, I think?

**Review by BartMassey — CHANGES_REQUESTED** [review]
Looks good at a quick read!  It's a lot of code, and I can't say I've read it carefully enough to be 100% confident, but I didn't see anything obviously wrong. Fix a couple of (I think?) wrong Issue references and I think it's good to go.

Thanks much: this turned out to be a ton of work for you. I'm sorry. :slightly_frowning_face:

File: CHANGELOG.md (hunk 1 above)
Comment (dhardy) [reply CHANGELOG.md]: I usually reference the PR :shrug:

**Review by BartMassey — APPROVED** [review]

**Pull request merged**